### PR TITLE
ceph-common: fix ceph options default path again

### PR DESCRIPTION
**Pull request by leseb**
closes: #788

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
this is please

**Comment by leseb** [comment]
test this please

**Comment by andrewschoen** [comment]
Looks good to me. One small nitpick, we might benefit from a comment briefly explaining why we need to do this.

**Comment by leseb** [comment]
@andrewschoen done thanks for the review!

File: roles/ceph-common/tasks/main.yml [hunk 1]
```diff
@@ -231,10 +231,35 @@
     line: "CLUSTER={{ cluster }}"
   when: ansible_os_family == "RedHat"
 
+# NOTE(leseb): we are performing the following check
+# to ensure any Jewel installation will not fail.
+# The following commit https://github.com/ceph/ceph/commit/791eba81a5467dd5de4f1680ed0deb647eb3fb8b
+# fixed a package issue where the path was the wrong.
+# This bug is not yet on all the distros package so we are working around it
```
Comment (ktdreyer) [inline roles/ceph-common/tasks/main.yml]: Please list the specific versions here so we know who/where to follow up.

Comment (smatzek) [inline roles/ceph-common/tasks/main.yml]: As noted in the bug, I believe this bug also affects installing previous versions including Infernalis, Hammer, etc.

Comment (leseb) [inline roles/ceph-common/tasks/main.yml]: Done

**Comment by smatzek** [comment]
@ktdreyer The Ubuntu bug asking for the backport of the ceph change to the appropriate Ubuntu deb packages is https://bugs.launchpad.net/ubuntu/+source/ceph/+bug/1582773

**Comment by leseb** [comment]
@ktdreyer @andrewschoen guys, is my note enough or do you need more?

**Comment by leseb** [comment]
test this please

File: roles/ceph-common/tasks/main.yml (hunk 1 above)
Comment (ktdreyer) [inline roles/ceph-common/tasks/main.yml]: Please add https://bugs.launchpad.net/ubuntu/+source/ceph/+bug/1582773 

I'm not sure about Infernalis from Ceph.com, but Hammer from ceph.com is not affected. This is why we've got to have confirmed tickets.

Comment (leseb) [inline roles/ceph-common/tasks/main.yml]: Ok will update and wait for a confirmation about infernalis ceph.com

Comment (smatzek) [inline roles/ceph-common/tasks/main.yml]: The commit that injected the error in Ceph is listed as being in Inferalis but not in Hammer [1].
I have not been able to check if the fix has been cherry picked from Ceph master into Inferalis.

[1] https://github.com/ceph/ceph/commit/7384a14f243519547a61534d22e21e6069aae016

**Comment by leseb** [comment]
@andrewschoen @ktdreyer  are we good with this one?

**Comment by andrewschoen** [comment]
:thumbsup: Looks good to me

File: roles/ceph-common/tasks/main.yml (hunk 1 above)
Comment (ktdreyer) [inline roles/ceph-common/tasks/main.yml]: What is the ticket URL for fixing this in Infernalis?

**Comment by andrewschoen** [comment]
test this please

File: roles/ceph-common/tasks/main.yml (hunk 1 above)
Comment (leseb) [inline roles/ceph-common/tasks/main.yml]: dunno

Comment (smatzek) [inline roles/ceph-common/tasks/main.yml]: @ktdreyer @leseb I can open a ticket for the Infernalis backport if that is holding this up.  However, I question whether we need it on Infernalis if it is going to be retired in a few weeks, planned June 2016.

File: roles/ceph-common/tasks/main.yml [hunk 2]
```diff
@@ -236,6 +236,11 @@
 # The following commit https://github.com/ceph/ceph/commit/791eba81a5467dd5de4f1680ed0deb647eb3fb8b
 # fixed a package issue where the path was the wrong.
 # This bug is not yet on all the distros package so we are working around it
+# Impacted versions:
+# - Jewel from UCA: https://bugs.launchpad.net/ubuntu/+source/ceph/+bug/1582773
+# - Jewel from latest Canonical 16.04 distro
+# - All previous versions from Canonical
```
Comment (ktdreyer) [inline roles/ceph-common/tasks/main.yml]: What does "All previous versions from Canonical" mean?

If I was maintaining this Ansible code a year from now and I read that comment, the phrasing would make it unclear if we can ever revert this.

File: roles/ceph-common/tasks/main.yml (hunk 1 above)
Comment (ktdreyer) [inline roles/ceph-common/tasks/main.yml]: Ok, that works for me re: Infernalis.

I can't wait to pull out all the Infernalis-related code.

**Pull request merged**